src/nanoarrow/common/array_test.cc

```diff
@@ -98,9 +98,11 @@ TEST(ArrayTest, ArrayTestAllocateChildren) {
   ArrowArrayRelease(&array);
 
   ASSERT_EQ(ArrowArrayInitFromType(&array, NANOARROW_TYPE_STRUCT), NANOARROW_OK);
+#if !defined(__SANITIZE_ADDRESS__)
```

Comment:
I did also explicitly add in this code that takes out blocks which are known to be flagged by the sanitizers. This is a bit of "extra credit" but I think generally safer than ignoring a broad class of potential errors.

If we didn't want to use these macros I'm happy to revert. Otherwise, we could look at a suppression file instead of changes to the code:

https://clang.llvm.org/docs/AddressSanitizer.html#issue-suppression

Reply:
Nevermind figured this out - all green